contracts/MarginCalculator.sol

```diff
@@ -43,19 +45,288 @@ contract MarginCalculator {
         uint256 longExpiryTimestamp;
         uint256 longCollateralDecimals;
         uint256 collateralDecimals;
+        uint256 vaultType;
         bool isShortPut;
         bool isLongPut;
         bool hasLong;
         bool hasShort;
         bool hasCollateral;
     }
 
+    /// @dev oracle deviation value (1e27)
+    uint256 internal oracleDeviation;
+
+    /// @dev FixedPoint 0
+    FPI.FixedPointInt internal ZERO = FPI.fromScaledUint(0, BASE);
+
+    /// @dev mapping to store dust amount per option collateral asset (scaled by collateral decimals)
+    mapping(address => uint256) internal dust;
+
+    /// @dev mapping to store array of time to expiry per product
+    mapping(bytes32 => uint256[]) internal productTimeToExpiry;
```

Comment:
Think we should brainstorm variable names for this and timeToExpiryValue- think is a bit confusing.